### PR TITLE
feat(cli/ssh): implement wait options and deprecate no-wait

### LINKED ISSUE #7768: Add `--wait` option to `coder ssh`
Currently `coder ssh` supports `--no-wait`, however, there is no way for a user to wait if they want to. To fix this, we will add `--wait` (and accompanying env) to `coder ssh`.

- [ ] Add `--wait` flag that always waits for startup script
- [ ] Make `--wait` the primary flag (default `--wait=false`), `--no-wait=true` is an alias for `--wait=false` and takes precedence (if possible, let flag order decide precedence)

### PR DESCRIPTION
**Pull request by mafredri**
Fixes #7768
Refs #7893

### CODE REVIEW

File: cli/portforward.go [hunk 1]
```diff
@@ -92,6 +92,7 @@ func (r *RootCmd) portForward() *clibase.Cmd {
 				Fetch: func(ctx context.Context) (codersdk.WorkspaceAgent, error) {
 					return client.WorkspaceAgent(ctx, workspaceAgent.ID)
 				},
+				Wait: false,
```
Comment (mafredri) [inline cli/portforward.go:95]: Review: Before change from `NoWait -> Wait`, these were bugs, added the explicit false here to signal intent.

File: cli/ssh.go [hunk 2]
```diff
@@ -42,13 +42,15 @@ var (
...
+//nolint:gocyclo
 func (r *RootCmd) ssh() *clibase.Cmd {
 	var (
 		stdio          bool
 		forwardAgent   bool
 		forwardGPG     bool
 		identityAgent  string
 		wsPollInterval time.Duration
+		wait           bool
 		noWait         bool
```
Comment (johnstcn) [inline cli/ssh.go:54]: Same as with https://github.com/coder/coder/pull/7893#discussion_r1221737760 - we could potentially (breaking-ly) refactor this to `--wait=(always|never|auto)`

**Review by Emyrk — APPROVED** [review]
I think this is a very clean solution. 

should we use an actual enum type for these? Not a big deal though

File: cli/ssh.go [hunk 3]
```diff
@@ -347,11 +380,13 @@ func (r *RootCmd) ssh() *clibase.Cmd {
...
 			Value:       clibase.DurationOf(&wsPollInterval),
 		},
+		waitOption,
 		{
 			Flag:        "no-wait",
 			Env:         "CODER_SSH_NO_WAIT",
-			Description: "Specifies whether to wait for a workspace to become ready before logging in (only applicable when the startup script behavior is blocking). Note that the workspace agent may still be in the process of executing the startup script and the workspace may be in an incomplete state.",
+			Description: "Enter workspace immediately after the agent has connected. This is the default if the template has configured the agent startup script behavior as non-blocking.",
 			Value:       clibase.BoolOf(&noWait),
+			UseInstead:  []clibase.Option{waitOption},
```
Comment (johnstcn) [inline cli/ssh.go:389]: It took me a hot second to see this as I was looking for `Deprecated` 🙃

**Review by johnstcn — APPROVED** [review]

**Pull request merged**